src/test/groovy/pl/tomaszdziurko/jvm_bloggers/mailing/BlogPostForMailItemSpec.groovy

```diff
@@ -28,10 +29,23 @@ class BlogPostForMailItemSpec extends Specification {
                 getBlog() >> author
             }
         when:
-            BlogPostForMailItem blogPostForMailItem = new BlogPostForMailItem(post)
+            BlogPostForMailItem blogPostForMailItem = BlogPostForMailItem.builder().from(post).build()
         then:
             blogPostForMailItem.authorLabel == "<a href=\"https://twitter.com/" + twitter.substring(1) + "\">" + name + "</a>"
     }
+    
+    def "Should build URL with default UTM parameters"() {
+        given:
+            Blog author = stubAuthorWith("Jan Kowalski", "@JanKowalski")
+            BlogPost post = Stub(BlogPost) {
+                getBlog() >> author
+                getUrl() >> "http://www.blog.pl"
+            }
+        when:
+            BlogPostForMailItem blogPostForMailItem = BlogPostForMailItem.builder().from(post).withDefaultUTMParameters().build()
+        then:
+            blogPostForMailItem.url == "http://www.blog.pl?utm_source=jvmbloggers&utm_medium=newsletter&utm_campaign=jvmbloggers#" + new SimpleDateFormat("yyyy-mm-dd").format(new Date());
```

Comment:
Occasionally (just before/after midnight) this could fail :) That is why we should use mocked NowProvider for such cases :)